url882f91ec97_WenboCao_Microsoft_Drone_tgz-pJ8-org_ejml_alg_dense_decomposition_qr_StabilityQRDecompositionJ8/src/org/ejml/ops/TestCommonOps.java

```diff
@@ -0,0 +1,900 @@
+                tranA = true;
+                tranB = true;
+            } else if( name.contains("TransA")) {
+                tranA = true;
+            } else if( name.contains("TransB")) {
+                tranB = true;
+            }
+//            System.out.println("Function = "+name+"  alpha = "+hasAlpha);
+
+            try {
+                if( !checkMultMethod(method,checkMethod,hasAlpha,tranA,tranB) ) {
+                    System.out.println("Failed: Function = "+name+"  alpha = "+hasAlpha);
+                    oneFailed = true;
+                }
+            } catch (InvocationTargetException e) {
+                throw new RuntimeException(e);
+            } catch (IllegalAccessException e) {
+                throw new RuntimeException(e);
+            }
+            numChecked++;
+        }
+        assertEquals(16,numChecked);
+        assertTrue(!oneFailed);
+    }
+
+    private Method findCheck( String name , boolean hasAlpha ) {
+        Method checkMethod;
+        try {
+            if( hasAlpha )
+                checkMethod = MatrixMatrixMult.class.getMethod(
+                        name,double.class,
+                        RowD1Matrix64F.class, RowD1Matrix64F.class,RowD1Matrix64F.class);
+            else
+                checkMethod = MatrixMatrixMult.class.getMethod(
+                        name, RowD1Matrix64F.class, RowD1Matrix64F.class,RowD1Matrix64F.class);
+        } catch (NoSuchMethodException e) {
+            checkMethod = null;
+        }
+        if( checkMethod == null ) {
+            try {
+            if( hasAlpha )
+                checkMethod = MatrixMatrixMult.class.getMethod(
+                        name+"_reorder",double.class,
+                        RowD1Matrix64F.class, RowD1Matrix64F.class,RowD1Matrix64F.class);
+            else
+                checkMethod = MatrixMatrixMult.class.getMethod(
+                        name+"_reorder", RowD1Matrix64F.class, RowD1Matrix64F.class,RowD1Matrix64F.class);
+            } catch (NoSuchMethodException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return checkMethod;
+    }
+
+    private boolean checkMultMethod(Method method, Method checkMethod, boolean hasAlpha,
+                                    boolean tranA, boolean tranB ) throws InvocationTargetException, IllegalAccessException {
+
+
+        // check various sizes
+        for( int i = 1; i < 40; i++ ) {
+            DenseMatrix64F a;
+            if( tranA ) a = RandomMatrices.createRandom(i+1,i,rand);
+            else  a = RandomMatrices.createRandom(i,i+1,rand);
+
+            DenseMatrix64F b;
+            if( tranB ) b = RandomMatrices.createRandom(i,i+1,rand);
+            else  b = RandomMatrices.createRandom(i+1,i,rand);
+
+            DenseMatrix64F c = RandomMatrices.createRandom(i,i,rand);
+            DenseMatrix64F c_alt = c.copy();
+
+            if( hasAlpha ) {
+                method.invoke(null,2.0,a,b,c);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)